### PR TITLE
Adds a customizable "seed" to posibrains to tell ghosts what the posibrain will be used for

### PR DESCRIPTION
**Pull request by XDTM**
## About The Pull Request

As the title says.
The seed can be set with Alt-click, as explained in an examine message.

This is _not_ meant to be an objective, just a kind of advertisement.

Also codedoc'd some of the code while i was there.

posibrain

## Why It's Good For The Game
Sometimes someone might make positronic brains for some interesting, unique and/or nonstandard use, and ghosts might ignore it assuming they're end up as a cyborg as usual, or enter the brain with the same assumption and be let down when the posibrain is instead made into an AI, or implanted in someone's chest cavity.

So this lets ghosts know in advance what they're in for.

## Changelog
:cl: XDTM
add: Positronic brains can now be given a "personality seed" by Alt-clicking them. Effectively, this tells ghosts what you plan on using the posibrain for, to advertise and prevent misunderstandings.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
### Body created
positronic brain requested in Robotics Lab! Personality requested: [MALF BORG FREE ANTAG]

File: code/modules/mob/living/brain/posibrain.dm [hunk 1]
```diff
@@ -6,37 +6,47 @@ GLOBAL_VAR(posibrain_notify_cooldown)
 	icon = 'icons/obj/assemblies.dmi'
 	icon_state = "posibrain"
 	w_class = WEIGHT_CLASS_NORMAL
-	var/next_ask
-	var/askDelay = 600 //one minute
+	var/ask_role = "" ///Can be set to tell ghosts what the brain will be used for
```
Comment (Fikou) [inline code/modules/mob/living/brain/posibrain.dm]: i think you need to put autodoc comments before the var or else it gives the autodoc to the var after it

File: code/modules/mob/living/brain/posibrain.dm [hunk 2]
```diff
@@ -57,6 +67,16 @@ GLOBAL_VAR(posibrain_notify_cooldown)
 	update_icon()
 	addtimer(CALLBACK(src, .proc/check_success), askDelay)
 
+/obj/item/mmi/posibrain/AltClick(mob/living/user)
+	if(!istype(user) || !user.canUseTopic(src, BE_CLOSE))
+		return
+	var/input_seed = stripped_input(user, "Enter a personality seed", "Enter seed", ask_role)
```
Comment (Tlaltecuhtli) [inline code/modules/mob/living/brain/posibrain.dm]: is there a max string lenght for this

Comment (XDTM) [reply code/modules/mob/living/brain/posibrain.dm]: The default is 1024, but on second thought it's probably too much

File: code/modules/mob/living/brain/posibrain.dm (hunk 1 above)
Comment (XDTM) [reply code/modules/mob/living/brain/posibrain.dm]: a lot of vars so far are commented like this, if that's actually the case there's a lot of fixing to do

Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/brain/posibrain.dm]: https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md#documenting-a-variable

File: code/modules/mob/living/brain/posibrain.dm (hunk 2 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/brain/posibrain.dm]: just make it name length which is ~48.

"Paladin AI" / "Malf AI" / "AI with custom laws"/ "gygax with weapons"

We also don't want to spam the popupboxes / chat with absurdly long messages.

File: code/modules/mob/living/brain/posibrain.dm (hunk 1 above)
Comment (XDTM) [reply code/modules/mob/living/brain/posibrain.dm]: Does that mean that there is indeed a lot of fixing to do?

Comment (kriskog) [reply code/modules/mob/living/brain/posibrain.dm]: It pulls from the line before afaik, so yeah. Either way, put them above for consistency with the rest of the codebase. Note, this applies to the /// autodoc comments, not the // non autodoc comments.

**Pull request merged**